frontend/components/canvas.tsx

```diff
@@ -1,75 +1,85 @@
-"use client";
+'use client'
 
-import { useDraw } from "@/hooks/useDraw";
-import { useState } from "react";
-import { CompactPicker } from "react-color";
+import { FC, useEffect, useState } from 'react'
+import { useDraw } from '../hooks/useDraw'
+import { ChromePicker } from 'react-color'
 
-export default function Canvas() {
-  const [color, setColor] = useState<string>('#000');
-  const [width, setWidth] = useState<number>(5);
-  const { canvasRef, onMouseDown, clear } = useDraw(drawLine);
+import { io } from 'socket.io-client'
+import { drawLine } from '../util/drawLine'
+const socket = io('http://localhost:3001')
 
-  function drawLine({ prevPoint, currentPoint, context }: Draw) {
-    const { x: currX, y: currY } = currentPoint;
-    const lineColor = color;
-    const lineWidth = width;
+interface pageProps {}
 
-    let startPoint = prevPoint ?? currentPoint;
-    context.beginPath();
-    context.lineWidth = lineWidth;
-    context.strokeStyle = lineColor;
-    context.moveTo(startPoint.x, startPoint.y);
-    context.lineTo(currX, currY);
-    context.stroke();
+type DrawLineProps = {
+  prevPoint: Point | null
+  currentPoint: Point
+  color: string
+}
+
+const page: FC<pageProps> = () => {
+  const [color, setColor] = useState<string>('#000')
+  const { canvasRef, onMouseDown, clear } = useDraw(createLine)
+
+  useEffect(() => {
+    const ctx = canvasRef.current?.getContext('2d')
+
+    socket.emit('client-ready')
+
+    socket.on('get-canvas-state', () => {
+      if (!canvasRef.current?.toDataURL()) return
+      console.log('sending canvas state')
+      socket.emit('canvas-state', canvasRef.current.toDataURL())
+    })
+
+    socket.on('canvas-state-from-server', (state: string) => {
```

Comment:
Need to implement the backend portion to send the current canvas state on the backend. 

In `backend\src\socket.ts`, see the `joinRoom` socket event handler. When a player joins a room, we will need to emit the canvas state event you created here and send it to the person who just joined so they receive the current canvas state and can see what has been drawn so far

so `io.to(socket.id).emit('canvas-state-from-server', <state of canvas>)`